perfscale_regression_ci/scripts/scalability/node-affinity-anti-affinity.sh

```diff
@@ -0,0 +1,171 @@
+# oc label nodes ${node_array[2]} cpu-
+# oc label nodes ${node_array[1]} --overwrite ${initial_node_label}
+
+
+echo -e "\nLabeling node ${node_array[1]} with label 'cpu=4'"
+oc label nodes ${node_array[1]} cpu=4
+
+echo -e "\nLabeling node ${node_array[2]} with label 'cpu=6'"
+oc label nodes ${node_array[2]} cpu=6
+
+echo -e "\nLabeling node ${node_array[1]} with label 'beta.kubernetes.io/arch=intel'"
+oc label nodes ${node_array[1]} --overwrite beta.kubernetes.io/arch=intel
+
+show_node_labels
+
+sleep 5
+
+
+echo "======Use kube-burner to load the cluster with test objects======"
+run_workload
+
+
+echo "======Waiting for the pods in both namespaces======"
+sleep 60
+sleep 30
+
+
+echo "======Checking the pods for errors======"
+
+check_no_error_pods $AFFINTIY_NAMESPACE
+
+check_no_error_pods $ANTI_AFFINTIY_NAMESPACE
+
+
+echo "======Counting Pods in each namespace======"
+node_affinity_pods_expected=$AFFINITY_JOB_ITERATION
+node_anti_affinity_pods_expected=$ANTI_AFFINITY_JOB_ITERATION
+
+echo "nodes ${node_array}"
+
+node_affinity_pods_actual=$(oc get pods -n node-affinity-0 -o wide | grep "node-affinity" | grep ${node_array[2]} | grep Running | wc -l | xargs )
+
+node_anti_affinity_pods_actual=$(oc get pods -n node-anti-affinity-0 -o wide | grep "hello-pod-anti-affinity" | grep -v ${node_array[2]} | grep Running | wc -l | xargs)
+
+
+echo "======Compare the expected and actual number of pods for each namespace. Get the PASS/FAIL result for each namespace======"
+pass_or_fail=0
+
+echo "node_affinity_pods_expected $node_affinity_pods_expected"
+
+if [ $node_affinity_pods_expected == $node_affinity_pods_actual ]; then
+  echo -e "Actual $node_affinity_pods_actual pods were sucessfully deployed. Node affinity test passed!"
+  (( ++pass_or_fail ))
+else
+  echo -e "Actual $node_affinity_pods_actual pods deployed does NOT match expected $node_affinity_pods_expected pods for node affinity test.  Node affinity test failed !"
+fi
+
+
+echo "node_anti_affinity_pods_expected $node_anti_affinity_pods_expected"
+
+if [ $node_anti_affinity_pods_expected == $node_anti_affinity_pods_actual ]; then
+  echo -e "Actual $node_anti_affinity_pods_actual pods were sucessfully deployed.  Node Anti-affinity test passed!"
+  (( ++pass_or_fail ))
+else
+  echo -e "Actual $node_anti_affinity_pods_actual pods deployed does NOT match expected $node_anti_affinity_pods_expected pods for node Anti-affinity test. Node Anti-affinity test failed !"
+fi
+
+sleep 60
+
+
+echo "======Final test result======"
+if [[ ${pass_or_fail} == 2 ]]; then
+  echo -e "\nOverall Node Affinity and Anti-affinity Testcase result:  PASS"
+  echo "======Clean up test environment======"
+  # # delete projects:
+  # ######### Clean up: delete projects and wait until all projects and pods are gone
+  echo "Deleting test objects"
+  delete_project_by_label kube-burner-job=$AFFINTIY_NAME
+  delete_project_by_label kube-burner-job=$ANTI_AFFINTIY_NAME
+
+  sleep 30
```

Comment:
don't think we need this sleep here, the delete_project_by_label will take care of the waiting till the project is gone